pykechain/models/activity2.py

```diff
@@ -920,3 +921,110 @@ def move(self, parent, classification=None):
         :raises APIError: if an Error occurs.
         """
         return self._client.move_activity(self, parent, classification=classification)
+
+    def share_link(self, subject=None, message=None, recipient_users=None):
+        """
+        Share the link of the `Activity` through email.
+
+        :param subject: subject of email
+        :type subject: basestring
+        :param message: message of email
+        :type message: basestring
+        :param recipient_users: users that will receive the email
+        :type recipient_users: list(Union(User, Id))
+        :raises IllegalArgumentError: if no 'message' is specified
+        :raises IllegalArgumentError: if no 'subject' is specified
+        :raises IllegalArgumentError: if no 'recipient_users' is specified
+        :raises APIError: if an Error occurs.
+        """
+        if not message:
+            raise IllegalArgumentError('Sharing an activity link requires a message')
+        if not subject:
+            raise IllegalArgumentError('Sharing an activity link requires a subject')
+        if not recipient_users:
+            raise IllegalArgumentError('Sharing an activity link requires recipient users')
+        else:
+            recipient_users_ids = list()
+            for user in recipient_users:
+                if isinstance(user, User):
+                    recipient_users_ids.append(user.id)
+                else:
+                    recipient_users_ids.append(user)
+
+        params = dict(
+            message=message,
+            subject=subject,
+            recipient_users=recipient_users_ids,
+            activity_id=self.id
+        )
+
+        url = self._client._build_url('notification_share_activity_link')
+
+        response = self._client._request('POST', url, data=params)
+
+        if response.status_code != requests.codes.created:  # pragma: no cover
+            raise APIError("Could not share the link to Activity, {}:\n\n{}'".format(str(response), response.json()))
+
+    def share_pdf(self, subject=None, message=None, recipient_users=None, paper_size=PaperSize.A4,
```

Comment:
Same as earlier comment: if a value must be provided a default value is not advised

Reply:
```suggestion
    def share_pdf(self, subject, message, recipient_users, paper_size=PaperSize.A3,
```